wagtail/core/tests/test_audit_log.py

```diff
@@ -0,0 +1,274 @@
+        self.assertEqual(PageLogEntry.objects.get_for_user(self.user).count(), 1)  # the create from setUp
+
+
+class TestAuditLog(TestCase):
+    def setUp(self):
+        self.root_page = Page.objects.get(id=1)
+
+        self.home_page = self.root_page.add_child(
+            instance=SimplePage(title="Homepage", slug="home2", content="hello")
+        )
+
+    def test_page_create(self):
+        self.assertEqual(PageLogEntry.objects.count(), 1)  # homepage
+
+        page = self.home_page.add_child(
+            instance=SimplePage(title="Hello", slug="my-page", content="world")
+        )
+        self.assertEqual(PageLogEntry.objects.count(), 2)
+        log_entry = PageLogEntry.objects.order_by('pk').last()
+        self.assertEqual(log_entry.action, 'wagtail.create')
+        self.assertEqual(log_entry.page_id, page.id)
+        self.assertEqual(log_entry.content_type, page.content_type)
+        self.assertEqual(log_entry.label, page.get_admin_display_title())
+
+    def test_page_edit(self):
+        # Directly saving a revision should not yield a log entry
+        self.home_page.save_revision()
+        self.assertEqual(PageLogEntry.objects.count(), 1)
+
+        # Explicitly ask to record the revision change
+        self.home_page.save_revision(log_action=True)
+        self.assertEqual(PageLogEntry.objects.count(), 2)
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.create').count(), 1)
+
+        # passing a string for the action should log this.
+        self.home_page.save_revision(log_action='wagtail.custom_action')
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.custom_action').count(), 1)
+
+    def test_page_publish(self):
+        revision = self.home_page.save_revision()
+        revision.publish()
+        self.assertEqual(PageLogEntry.objects.count(), 2)
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.publish').count(), 1)
+
+    def test_page_rename(self):
+        # Should not log a name change when publishing the first revision
+        revision = self.home_page.save_revision()
+        self.home_page.title = "Old title"
+        self.home_page.save()
+        revision.publish()
+
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.publish').count(), 1)
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.rename').count(), 0)
+
+        # Now, check the rename is logged
+        revision = self.home_page.save_revision()
+        self.home_page.title = "New title"
+        self.home_page.save()
+        revision.publish()
+
+        self.assertEqual(PageLogEntry.objects.count(), 4)
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.publish').count(), 2)
+        self.assertEqual(PageLogEntry.objects.filter(action='wagtail.rename').count(), 1)
+
+    def test_page_unpublish(self):
+        self.home_page.unpublish()
+        self.assertEqual(PageLogEntry.objects.count(), 2)
```

Comment:
Could we add an assertion for the other log entry as well?

Reply:
If this other log entry is created by `.add_child` in the `setUp()`, maybe we should delete all log entries at the end of `setUp()` so only relevant entries exist?